power-control-x86/src/power_control.cpp

```diff
@@ -144,18 +163,57 @@ static int getGPIOValue(const std::string& name)
     return value;
 }
 
+static void setFrontpanelPowerLed(bool led)
+{
+    int fd = -1;
+    char i2c_devname[FILEPATHSIZE];
+    int portRead, portWrite, dataRead, dataWrite;
+
+    // check volcano FP
+    snprintf(i2c_devname, FILEPATHSIZE, "/dev/i2c-%d", FP_I2C_BUS1);
+    fd = open(i2c_devname, O_RDWR);
+    if(fd < 0) {
+        // check Purico FP
+        snprintf(i2c_devname, FILEPATHSIZE, "/dev/i2c-%d", FP_I2C_BUS2);
+        fd = open(i2c_devname, O_RDWR);
+        if(fd < 0)
+            return; // cannot open i2c bus
+    }
+
+    if (ioctl(fd, I2C_SLAVE, FP_IOX_ADDR) >= 0) {
+        // read IOX Port Reg
+        portRead = i2c_smbus_read_byte_data(fd, FP_IOX_PORT_REG);
+        dataRead = i2c_smbus_read_byte_data(fd, FP_IOX_DATA_REG);
+        portWrite = portRead & FP_POWER_PORT_MASK;
+        if (led == true)
+            dataWrite = dataRead & FP_POWER_ON_REG_MASK;
+        else
+            dataWrite = dataRead | FP_POWER_OFF_REG_MASK;
+        i2c_smbus_write_byte_data(fd, FP_IOX_PORT_REG, portWrite);
+        i2c_smbus_write_byte_data(fd, FP_IOX_DATA_REG, dataWrite);
+    }
+    else {
+        std::cerr << "Error ioctl call for Front Panel i2c device " << FP_IOX_ADDR << " \n";
+    }
+    if (fd >= 0) {
+        close(fd);
+    }
+}
+
 static PowerState powerState;
 static std::string getPowerStateName(PowerState state)
 {
     switch (state)
     {
         case PowerState::on:
+            setFrontpanelPowerLed(true);
```

Comment:
Check power on/off multiple times as the app is time sensitive

Reply:
done. in both Purico and Volcano